src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -544,6 +548,93 @@ bool process_subgraph(ov::pass::ModelPass& model_pass, const std::shared_ptr<Nod
     return changed;
 }
 
+static std::string ParseSymbolVariant(std::vector<symbol_variant> values) {
+    std::vector<std::string> symbol_strings;
+    symbol_strings.reserve(values.size());
+    for (auto& value : values) {
+        if (std::holds_alternative<float>(value)) {
+            symbol_strings.push_back(std::to_string(std::get<float>(value)));
+        } else if (std::holds_alternative<int>(value)) {
+            symbol_strings.push_back(std::to_string(std::get<int>(value)));
+        } else if (std::holds_alternative<int32_t>(value)) {
+            symbol_strings.push_back(std::to_string(std::get<int32_t>(value)));
+        } else if (std::holds_alternative<int64_t>(value)) {
+            symbol_strings.push_back(std::to_string(std::get<int64_t>(value)));
+        } else {
+            symbol_strings.push_back(std::get<std::string>(value));
+        }
+    }
+
+    return ov::util::join(symbol_strings);
+}
+
+std::shared_ptr<ov::Node> NewGenStridedSlice(const std::shared_ptr<ov::Node>& data,
+                                             const ov::pass::pattern::PatternOp& start,
+                                             const ov::pass::pattern::PatternOp& stop,
+                                             const ov::pass::pattern::PatternOp& step,
+                                             size_t axis) {
+    std::vector<int64_t> begin_mask(axis + 1, 1);
+    std::vector<int64_t> end_mask(axis + 1, 1);
+    std::vector<int64_t> new_axis_mask;
+    std::vector<int64_t> shrink_axis_mask;
+    std::vector<int64_t> ellipsis_mask;
+
+    begin_mask[axis] = 0;
+    end_mask[axis] = 0;
+
+    return ov::pass::pattern::wrap_type<ov::op::v1::StridedSlice>({data, start, stop, step},
+                                                                  {{"begin_mask", begin_mask},
+                                                                   {"end_mask", end_mask},
+                                                                   {"new_axis_mask", new_axis_mask},
+                                                                   {"shrink_axis_mask", shrink_axis_mask},
+                                                                   {"ellipsis_mask", ellipsis_mask}});
+}
+
+std::shared_ptr<ov::Node> NewGenSlice(const std::shared_ptr<ov::Node>& data,
+                                      symbol_variant start,
+                                      symbol_variant stop,
+                                      symbol_variant step,
+                                      size_t axis) {
+    using namespace ov::pass;
+    auto slice_start = ParseSymbolVariant({start});
+    auto slice_stop = ParseSymbolVariant({stop});
+    auto slice_step = ParseSymbolVariant({step});
+    auto slice_axis = ParseSymbolVariant({static_cast<int64_t>(axis)});
+
```

Comment:
should we check here, that step is not 0?
 std::visit([](const auto& step_val) {
          if constexpr (std::is_same_v<std::decay_t<decltype(step_val)>, int64_t>) {
              if (step_val == 0) {
                  throw std::invalid_argument("Step cannot be zero in slice operation");
              }
          }
      }, step);